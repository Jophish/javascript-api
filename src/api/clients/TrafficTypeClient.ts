export default class TrafficTypeClient {
  
  list() {};
}