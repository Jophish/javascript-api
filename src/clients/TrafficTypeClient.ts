import Client from './Client';
import TrafficType, { ITrafficType } from '../dtos/TrafficType';

export default class TrafficTypeClient extends Client {
  /**
   * Lists all traffic types
   */
  list(): Promise<TrafficType[]> {
    return this.gateway.get('/trafficTypes').then((res: any) => {
      return res.map(e => new TrafficType(e));
    });
  };
}
