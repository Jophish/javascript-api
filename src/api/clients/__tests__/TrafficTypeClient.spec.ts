jest.mock('../../http/gateway');

import TrafficTypeClient from '../TrafficTypeClient';

test('should be a class (function)', () => {
  expect(typeof TrafficTypeClient).toBe('function');
});

test('instances should expose an API', () => {
  const trafficType = new TrafficTypeClient();

  expect(trafficType.list).toBeDefined();
});

test('it should be able return a promise resolving to list the environments', () => {
  const trafficType = new TrafficTypeClient();
  const listPromise = trafficType.list();

  expect(listPromise.then).toBeDefined();
  expect(typeof listPromise.then).toBe('function');  

  return listPromise.then((res) => {
    expect(res).toMatchSnapshot();
  });
});