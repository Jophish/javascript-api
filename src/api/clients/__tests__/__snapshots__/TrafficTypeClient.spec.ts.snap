// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should be able return a promise resolving to list the environments 1`] = `
Object {
  "objects": Array [
    Object {
      "displayAttributeId": "name",
      "id": "83094920-4491-11e7-84b7-126798e48ef6",
      "name": "user",
    },
    Object {
      "displayAttributeId": "name",
      "id": "73d49980-4498-11e7-84b7-126798e48ef6",
      "name": "machine",
    },
    Object {
      "displayAttributeId": "name",
      "id": "8ef11380-4491-11e7-84b7-126798e48ef6",
      "name": "tt_1",
    },
  ],
}
`;
